app/models/concerns/string_validation.rb

```diff
@@ -0,0 +1,37 @@
+module StringValidation
```

Comment:
I feel like the name of this module doesn't quite reflect what it does - maybe something like `GeneratesObjectKeys` or something?